### ISSUE TITLE
[Questions] Usage inside Xcode Project

### ISSUE THREAD

**Issue by o15a3d4l11s2** [kiliankoe/swift-outdated#3]
We have an iOS app that uses a combination of SPM and Cocoapods. Is there a way to use the swift command to check for new SPM packages? The SPM dependencies are added as described here: https://developer.apple.com/documentation/xcode/adding_package_dependencies_to_your_app, so there is no Package.swift file inside the project.

If I try to execute the command I get
```bash
Error: ShellOut encountered an error
Status code: 1
Message: "error: root manifest not found"
Output: ""
```

**Comment by kiliankoe** [comment]
Hey @o15a3d4l11s2 👋 

While Xcode projects don't have a manifest, they do have a Package.resolved file which contains all the necessary info for this tool. Sounds like a good idea, I'll see what I can do to support that as well 😊 

It would be very cool to be able to run this as a build phase and output warnings for outdated dependencies 👍

**Comment by o15a3d4l11s2** [comment]
I just found this article: https://samwize.com/2019/03/22/integrate-swift-package-manager-for-ios-and-macos-projects/ which describes it in a slightly different way - create a package with the dependencies. Then there would be a Package.swift file. I will give it a try and come back to you, @kiliankoe. But the idea of using the Package.resolved file sounds amazing!

**Comment by kiliankoe** [comment]
That's an interesting workaround! But since the majority of people will use Xcode's SwiftPM integration anyways for their dependencies, directly supporting that feels like a worthwhile endeavor in any way.

**Comment by o15a3d4l11s2** [comment]
Using the workaround is too much of a hustle for something that comes out of the box for Cocoapods and Carthage. Feel free to ping me if I can help in any way for developing this feature.

**Comment by kiliankoe** [comment]
I've just pushed a new version, I threw out the old implementation which also fixed some permission weirdness on fetching available git tags of dependencies. The new version only reads the Package.resolved file and outputs a table of outdated dependencies only. ~~Also there's a new flag `--xcode`, which can be used when running this from inside an Xcode Run Script Phase.~~ It automatically detects when running inside Xcode and emits warnings for outdated dependencies 😊 

The only downside is that git tags for all dependencies are fetched on every build, which might take a second or two. So depending on if that's an issue for you I'd suggest maybe running this manually every now and then.

<img width="247" alt="image" src="https://user-images.githubusercontent.com/2625584/104966116-6cedc400-59e0-11eb-9dc0-942f860e9e33.png">

**Comment by o15a3d4l11s2** [comment]
While installing I only faced 1 issue - I had to use a more recent version of `swift-argument-parser` (mine was resolved to 0.0.2). After setting the version to the latest (0.3.2) the build succeeded. The error thrown was because of the `version` parameter in `CommandConfiguration` in Outdated.swift.

I am very happy that you were open to implementing this new feature and that you made it so fast! Many thanks.
